packages/util/src/functions/runtime.ts

```diff
@@ -0,0 +1,12 @@
+export function shouldUseGlobalFetchAndWebSocket() {
+	// Browser env and deno when ran directly
+	if (typeof globalThis.process === 'undefined') {
+		return 'fetch' in globalThis && 'WebSocket' in globalThis;
```

Comment:
For bun, you could probably check for these two as well and remove the `globalThis.process` check. Bun implements `fetch` and `WebSocket` as globals